lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -439,11 +440,96 @@ def mock_run(args, **kwargs):
             m.setattr(subprocess, "run", mock_run)
             m.setattr(Tarball, "__init__", TestCacheManager.MockTarball.__init__)
             m.setattr(Controller, "__init__", TestCacheManager.MockController.__init__)
+            m.setattr(os.path, "getmtime", lambda self: 1681814586.1138802)
             tb = Tarball(tar, Controller(Path("/mock/archive"), cache, None))
             tb.unpack()
             assert call == ["tar", "find"]
             assert tb.unpacked == cache / "ABC" / tb.name
 
+    def test_cachemap_modification_time(self, monkeypatch):
+        """Test to build the cache map of the unpacked tarball"""
+        tar = Path("/mock/A.tar.xz")
+        cache = Path("/mock/.cache")
+        call = list()
+
+        def mock_run(args, **kwargs):
+            call.append(args[0])
+
+            tar_target = "--file=" + str(tar)
+            assert args[0] == "find" or args[0] == "tar" and tar_target in args
+            return subprocess.CompletedProcess(
+                args, returncode=0, stdout="Successfully Unpacked!", stderr=None
+            )
+
+        def mock_getmtime(path):
+            """Mock file/directory modification time"""
+            if path == Path("/mock/.cache/ABC/A"):
+                raise FileNotFoundError(
+                    f"[Errno 2] No such file or directory: '{path}'"
+                )
+
+        with monkeypatch.context() as m:
+            m.setattr(Path, "mkdir", lambda path, parents: None)
+            m.setattr(subprocess, "run", mock_run)
+            m.setattr(Tarball, "__init__", TestCacheManager.MockTarball.__init__)
+            m.setattr(Controller, "__init__", TestCacheManager.MockController.__init__)
+            m.setattr(os.path, "getmtime", mock_getmtime)
+            tb = Tarball(tar, Controller(Path("/mock/archive"), cache, None))
+            with pytest.raises(FileNotFoundError) as exc:
+                tb.unpack()
+            assert (
+                str(exc.value)
+                == "[Errno 2] No such file or directory: '/mock/.cache/ABC/A'"
+            )
+            assert call == ["tar", "find"]
+            assert tb.unpacked == cache / "ABC" / tb.name
```

Comment:
Why should `tb.unpacked` have what looks like a "good value" when the call to `unpack()` aborted with a `FileNotFoundError`?

Reply:
These two assertions were passed because the values they are asserting with are gathered before `FileNotFounderror` was raised. I should have been careful about sequence.

Reply:
It is a good idea to have in mind what "the interface" to the function-under-test is -- that should be the target of the test; asserting anything which isn't part of the interface (or which represents a behavior which doesn't matter in the specific scenario) is just a waste of time/effort/attention/CPU.  So, if the value of `tb.unpacked` at the end of the test doesn't actually matter, then we don't need the assertion.  (Whether the value of `call` matters or not is a trickier question...we want some assurance that we're getting the behavior that we're trying to drive in the CUT, but it might not matter in _this_ test, since we have other similar tests which already check this.)